apollo-backend/src/main/java/io/logz/apollo/controllers/HealthController.java

```diff
@@ -22,17 +25,25 @@
 
     private final KubernetesHealth kubernetesHealth;
     private final EnvironmentDao environmentDao;
+    private final SlaveService slaveService;
     private static final Logger logger = LoggerFactory.getLogger(HealthController.class);
 
     @Inject
-    public HealthController(KubernetesHealth kubernetesHealth, EnvironmentDao environmentDao) {
-        this.kubernetesHealth= requireNonNull(kubernetesHealth);
-        this.environmentDao= requireNonNull(environmentDao);
+    public HealthController(KubernetesHealth kubernetesHealth, EnvironmentDao environmentDao, SlaveService slaveService) {
+        this.kubernetesHealth = requireNonNull(kubernetesHealth);
+        this.environmentDao = requireNonNull(environmentDao);
+        this.slaveService = requireNonNull(slaveService);
     }
 
     @GET("/health")
     public void getHealth(Req req) {
+        Set<Integer> scopedEnvironments = slaveService.getScopedEnvironments();
         Map<Integer, Boolean> environmentsHealthMap = kubernetesHealth.getEnvironmentsHealthMap();
+        environmentsHealthMap.keySet()
+                             .stream()
+                             .filter(envId -> !scopedEnvironments.contains(envId))
+                             .forEach(environmentsHealthMap::remove);
```

Comment:
You cant remove things from the map you are working on currently.
You must pass by a "temp" variable.

Also, no tests here? This should have failed your build